app/forms/admin/claims/employment_history/destroy_employment_form.rb

```diff
@@ -0,0 +1,39 @@
+module Admin
+  module Claims
+    module EmploymentHistory
+      class DestroyEmploymentForm
+        include ActiveModel::Model
+        include ActiveModel::Attributes
+
+        attr_reader :claim
+
+        attr_accessor :deleted_by
+
+        attribute :employment_id
+
+        validates :deleted_by, presence: true
+
+        def initialize(claim, params: {})
+          @claim = claim
+
+          super(params)
+        end
+
+        def save!
+          eligibility = claim.eligibility
+
+          employment = eligibility.employment_history.find { |e| e.id == employment_id }
+
+          raise ActiveRecord::RecordInvalid unless employment
```

Comment:
activerecord `find` will raise `ActiveRecord::RecordNotFound` which may be sufficient for your needs?